app/api/v1/endpoints/recommendations.py

```diff
@@ -0,0 +1,43 @@
+from flask import request
+from flask_restplus import Resource
+from app.api.v1.api import api
+from app.api.v1.models.recommendations import recommendation, recommendation_context, recommendation_parser
+from app.recommender_engine_stub import RecommenderEngineStub
+from app.user_activity_gatherer import UserActivityGatherer
+from app.recommender_engine_stub import InvalidRecommendationPanelLocationError, InvalidRecommendationPanelVersionError
+
+recommendation_name_space = api.namespace("recommendations", "Endpoint used for getting recommendations")
+
+
+@api.errorhandler(InvalidRecommendationPanelVersionError)
+def handle_root_exception(error):
+    return {'message': error.message()}, 400
+
+
+@api.errorhandler(InvalidRecommendationPanelLocationError)
+def handle_root_exception(error):
+    return {'message': error.message()}, 400
+
+
+@recommendation_name_space.route("")
+class Recommendation(Resource):
+    @api.expect(recommendation_context, recommendation_parser, validate=True)
+    @api.marshal_with(recommendation)
+    def post(self):
+        """Returns list of ids of recommended scientific services"""
+        args = recommendation_parser.parse_args()
+        location = args.get("location", None)
+        version = args.get("version", None)
+        context = request.get_json()
+
+        full_context = {
+            "user_id": context["user_id"],
+            "timestamp": "time",
+            "filters": context.get("filters", None),
+            "search_phrase": context.get("search_phrase", None)
+        }
+
+        recommended_services_ids = RecommenderEngineStub.get_recommendations(context, location, version)
+        UserActivityGatherer.save_state_and_action(state=full_context, action=recommended_services_ids)
+
+        return {"recommended_services_ids": recommended_services_ids}
```

Comment:
Nesting resources is not the best option (unless you want to avoid some problems on the client's side - but it should not apply here).

If your endpoint should return list, then return it directly, without packing it into dictionary with single key.

**NOTE** In some cases you might want to do that, for example if in the future some other mechanisms (like pagination should be added) - but in that case the dictionary key under which data is stored should be the same across all endpoints - something like `data`.

Generally - unless it's unavoidable I would disadvise it.

Reply:
Yes, I definitely want to return just plain list but it's impossible to define such a corresponding api model in restplus lib - it demands structured hash and it's known issue so it's necessary.

Reply:
I would actually check https://github.com/python-restx/flask-restx which is a fork of restplus (which is no longer maintained) - it should be a drop in replacement (as it's a fork) and it actually supports lists (see this closed issue https://github.com/python-restx/flask-restx/issues/65)